src/common/components/markdownViewer/markdownViewerComponent.tsx

```diff
@@ -1,33 +1,110 @@
 import * as React from 'react';
-import { marksy } from 'marksy';
+import * as ReactDOM from 'react-dom';
+import { withRouter } from 'react-router';
+import { SOURCE_LINE_CLASSNAME,
+         getPixelOffsetForSourceLine,
+         getSourceLineForPixelOffset } from './syncScroll';
+import { CreateMarkdownRender, Mdr, MdrFactory,
+         MdrSetup, MdrOptions, MdrCodeStyle } from './render';
+import { throttle } from '../../../common/helper/limitExecution';
 
-const compile = marksy({
-  // TODO: extract into a new <DynamicLink /> that analyze "href" and render a <Link /> or <a />
-  a: ({ target, children, ...other }) => (
-    <a target="_self" {...other}>{children}</a>
-  ),
-});
+/**
+ * TODO:
+ * Custom rule to analyze "href" and render a <Link /> (react router) or <a />
+ */
 
-export interface MarkDownViewerComponentProps {
+interface MarkDownViewerComponentProps {
   content: string;
+  scrollSourceLine?: number;
+  onScrollSourceLine?: (sourceLine) => any;
+  className?: string;
+  location?: any; // Router HOC injected.
+  options?: MdrOptions;
+  codeStyle?: MdrCodeStyle;
 }
 
-const getMarkDownChildren = (content: string): React.ReactNode => {
-  let childrenComponent: React.ReactNode = null;
+class MarkDownViewer extends React.Component<MarkDownViewerComponentProps, {}> {
+  constructor(props) {
+    super(props);
+  }
 
-  if (content) {
-    childrenComponent = compile(content).tree;
+  private CreateMdrInstance = () => {
+    const setupParams: MdrSetup = {
+      routerLocation: this.props.location ? this.props.location.pathname : '',
+      options: this.props.options,
+      codeStyle: this.props.codeStyle,
+    };
+    return CreateMarkdownRender(setupParams);
   }
 
-  return childrenComponent;
-};
+  private mdr: Mdr = this.CreateMdrInstance();
+  private scrollableContainerRef: HTMLElement = null;
+  private allowNotifyScroll: boolean = true;
```

Comment:
It's a bit strange that this property isn't in state.

Reply:
Yes, you are right Dani, it is actually an internal state property. Do you want me to change it?

Reply:
It should be in state, we should avoid member variables whenver is possible

Reply:
this issue is pending

Reply:
**Done**. [Link to commit](https://github.com/Lemoncode/LeanMood/pull/168/commits/0b86b91f18f426363c4ca3879acc90dd1c589771)

**NOTE**: I moved `mdr` and `allowNotifyScroll` to state but left `scrollableContainerRef` as a class member.  I made this decision based on [React official docs](https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-class-component) regarding `ref` property. They literally say: "*Using the `ref` callback just to set a property on the class is a common pattern for accessing DOM elements*". 
It make sense not to store the `ref` in state because `ref` is a unique handler to an `HTMLElement`. And we are gonna use this handler to mutate its properties. This would violate state immutability rules.